### PR TITLE
Make recommend command ignore case for ingredients

### LINKED ISSUE #179: [PE-D][Tester E] Recommend feature should be case insensitive
e.g. in a recipe I have "bread" as an ingredient but in ingredients list I have it as "Bread", the recommendation does not realise it is the same ingredient and is neglected.

Using the above example, recommend doesn't work when "Bread" is in ingredients list:

But works only when "bread" is in ingredients list:

-------------
Labels: `severity.Medium` `type.FunctionalityBug`
original: naveen42x/ped#2

### LINKED ISSUE #203: [PE-D][Tester B] `recommend` command can be more practical
Recipe in store:
```
        Recipe: test
          Ingredients:
            - something 500ml
          Steps:
            1. cook
```
```
         Recipe: test2
          Ingredients:
            - 4 eggs
          Steps:
            1. cook
```

Ingredient in store:
```
        Here's all the ingredients you currently have:
        1. 30 eggs
        2. something
```

Using `recommend` would give:
```
        __________________________________________________________________________________
        Unfortunately I can't recommend you any recipes,
        because you lack the ingredients for any recipes in my database.
        Perhaps you should hit up the grocery store.
        It will do you some good to go outside once in a while.
        __________________________________________________________________________________
```

`recommend` seems to only looks for exact match in ingredients, when reality it might not be that way. Perhaps some changes can be made to `recommend` design to work better with user's needs.

-------------
Labels: `severity.Low` `type.FeatureFlaw`
original: Metanyu/ped#8

### PR DESCRIPTION
**Pull request by 3CCLY**
Fixes #179,
Fixes #203

Recommend command now reads ingredients in all lowercase, to resolve the previous issue of not matching ingredients from the inventory to the ingredients of the recipes just because a different case was used.

Also implemented gskang-22's duplicate ingredient fix to the edit command.

Updated the edit command sequence diagram to reflect the updated sequence too

### CODE REVIEW

**Review by Sukkaito — APPROVED** [review]
LGTM! I've mentioned a suggestion about tests below, hope to receive some insights! 👀

File: docs/puml/SequenceDiagrams/EditRecipes.puml [hunk 1]
```diff
@@ -13,12 +13,11 @@ skinparam {
 box Editing Recipes COLOR_PALE_PINK
 participant "YMFC" as YMFC <<class>>
 participant "Parser" as Parser <<class>>
-participant ":EditCommand" as EditCommand
+participant "command:EditCommand" as EditCommand
 participant ":Storage" as Storage
 participant ":RecipeList" as RecipeList
 participant ":Storage" as Storage
 participant ":UI" as UI
```
Comment (seanngja) [inline docs/puml/SequenceDiagrams/EditRecipes.puml:20]: Maybe can initialise `:Storage` as `storage:Storage` to show how the storage object is referenced by its name, and similarly for recipeList and ui. Also, storage seems to be initialised twice here.

File: src/test/java/ymfc/commands/RecommendCommandTest.java [hunk 2]
```diff
@@ -41,22 +41,22 @@ void setUp() {
...
-        recipe3 = new Recipe("Salted ABC Soup", ingredients3, steps);
+        steps.add("Cook abc in salted water for 76 days.");
+        ingredients1.add(new Ingredient("abc"));
+        recipe1 = new Recipe("abc", ingredients1, steps);
+        ingredients2.add(new Ingredient("abc"));
+        ingredients2.add(new Ingredient("water"));
+        recipe2 = new Recipe("abc soup", ingredients2, steps);
+        ingredients3.add(new Ingredient("abc"));
+        ingredients3.add(new Ingredient("water"));
+        ingredients3.add(new Ingredient("salt"));
```
[the remark this answers is not in the document]
Comment (3CCLY) [reply src/test/java/ymfc/commands/RecommendCommandTest.java:52]: Good suggestion, will update

File: src/main/java/ymfc/parser/Parser.java [hunk 3]
```diff
@@ -394,6 +394,23 @@ private static EditCommand getEditCommand(String args) throws InvalidArgumentExc
         String ingredString = m.group("ingreds");
         ArrayList<Ingredient> ingreds = null;
         if (ingredString != null) {
+            //@@author gskang
```
[the remark this answers is not in the document]
Comment (3CCLY) [reply src/main/java/ymfc/parser/Parser.java:397]: Will do

**Pull request merged**